File: src/net/bdew/lib/capabilities/Capabilities.scala
/*
 * Copyright (c) bdew, 2013 - 2017
 * https://github.com/bdew/bdlib
 *
 * This mod is distributed under the terms of the Minecraft Mod Public
 * License 1.0, or MMPL. Please check the contents of the license located in
 * http://bdew.net/minecraft-mod-public-license/
 */

package net.bdew.lib.capabilities

import net.minecraftforge.common.capabilities.{Capability, CapabilityInject}
import net.minecraftforge.fluids.capability.{IFluidHandler, IFluidHandlerItem}
import net.minecraftforge.items.IItemHandler

import scala.annotation.meta.setter

object Capabilities {
  @(CapabilityInject@setter)(classOf[IFluidHandler])
  var CAP_FLUID_HANDLER: Capability[IFluidHandler] = _

  @(CapabilityInject@setter)(classOf[IFluidHandlerItem])
  var CAP_FLUID_HANDLER_ITEM: Capability[IFluidHandlerItem] = _

  @(CapabilityInject@setter)(classOf[IItemHandler])
  var CAP_ITEM_HANDLER: Capability[IItemHandler] = _
}

File: src/main/scala/net/bdew/lib/capabilities/Capabilities.java
package net.bdew.lib.capabilities;

import net.minecraftforge.common.capabilities.Capability;
import net.minecraftforge.common.capabilities.CapabilityManager;
import net.minecraftforge.common.capabilities.CapabilityToken;
import net.minecraftforge.energy.IEnergyStorage;
import net.minecraftforge.fluids.capability.IFluidHandler;
import net.minecraftforge.fluids.capability.IFluidHandlerItem;
import net.minecraftforge.items.IItemHandler;

@SuppressWarnings("CanBeFinal")
public class Capabilities {
    public static Capability<IItemHandler> CAP_ITEM_HANDLER = CapabilityManager.get(new CapabilityToken<>() {
    });
    public static Capability<IFluidHandler> CAP_FLUID_HANDLER = CapabilityManager.get(new CapabilityToken<>() {
    });
    public static Capability<IFluidHandlerItem> CAP_FLUID_HANDLER_ITEM = CapabilityManager.get(new CapabilityToken<>() {
    });
    public static Capability<IEnergyStorage> CAP_ENERGY_HANDLER = CapabilityManager.get(new CapabilityToken<>() {
    });
}
